python/semantic_kernel/connectors/ai/anthropic/services/anthropic_chat_completion.py

```diff
@@ -0,0 +1,247 @@
+
+        if not settings.ai_model_id:
+            settings.ai_model_id = self.ai_model_id
+
+        settings.messages = self._prepare_chat_history_for_request(chat_history)
+        try:
+            async with self.async_client.messages.stream(**settings.prepare_settings_dict()) as stream:
+                author_role = None
+                metadata: dict[str, Any] = {"usage": {}, "id": None}
+                content_block_idx = 0
+                
+                async for stream_event in stream:
+                    if isinstance(stream_event, RawMessageStartEvent):
+                        author_role = stream_event.message.role
+                        metadata["usage"]["input_tokens"] = stream_event.message.usage.input_tokens
+                        metadata["id"] = stream_event.message.id
+                    elif isinstance(stream_event, (RawContentBlockDeltaEvent, RawMessageDeltaEvent)):
+                        yield [self._create_streaming_chat_message_content(stream_event, 
+                                                                           content_block_idx, 
+                                                                           author_role, 
+                                                                           metadata)]
+                    elif isinstance(stream_event, ContentBlockStopEvent):
+                        content_block_idx += 1
+
+        except Exception as ex:
+            raise ServiceResponseException(
+                f"{type(self)} service failed to complete the request",
+                ex,
+            ) from ex
+
+    def _create_chat_message_content(self, 
+                                     response: Message, 
+                                     content: TextBlock, 
+                                     response_metadata: dict[str, Any]) -> "ChatMessageContent":
+        """Create a chat message content object."""
+        items: list[ITEM_TYPES] = []
+        
+        if content.text:
+            items.append(TextContent(text=content.text))
+
+        finish_reason = None
+        if response.stop_reason:
+            finish_reason = ANTHROPIC_TO_SEMANTIC_KERNEL_FINISH_REASON_MAP[response.stop_reason]
+        
+        return ChatMessageContent(
+            inner_content=response,
+            ai_model_id=self.ai_model_id,
+            metadata=response_metadata,
+            role=AuthorRole(response.role),
+            items=items,
+            finish_reason=finish_reason,
+        )
+
+    def _create_streaming_chat_message_content(self, 
+                                               stream_event: RawContentBlockDeltaEvent | RawMessageDeltaEvent,
+                                               content_block_idx: int,
+                                               role: str | None = None,
+                                               metadata: dict[str, Any] = {}) -> StreamingChatMessageContent:
```

Comment:
Looks like auto formatting or commit hook is not set up in your dev environment? This looks like it should be formatted differently.

Reply:
commit hook did not flag this one, but i did just update it now to match the formatting of functions in other parts of the code base.